### PR TITLE
booking controller & Trafikverket API logic

### PR DESCRIPTION
**Pull request by adambrodin**
- Added a booking controller to able to search and retrieve available trips
- Service to talk with and handle data from Trafikverket API
- Parsing script to import TrainStation API data into the database
- Refactoring

### CODE REVIEW

File: api/src/controllers/booking.controller.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      return {
+        response: 'Trips have been fetched successfully.',
+        data: fetchedTrips.trips,
+      };
+    } else {
+      return { response: 'No trips were found for the given criteria.' };
+    }
+  }
+
+  @Post('book')
```
Comment (Spuute) [inline api/src/controllers/booking.controller.ts]: What do you think about changing 'book' to 'reservation' which is a noun?
@Post('reservation')

File: api/src/entities/ticket.entity.ts [hunk 2]
```diff
@@ -8,13 +8,7 @@ import {
 } from 'typeorm';
 import { BookingEntity } from './booking.entity';
 import { TrainEntity } from './train.entity';
-
-export enum TicketType {
-  ADULT = 'Adult',
-  STUDENT = 'Student',
-  PENSIONER = 'Pensioner',
```
Comment (Spuute) [inline api/src/entities/ticket.entity.ts:15]: What do you think about SENIOR instead of Pensioner?

File: shared/enums/ticket-type.enum.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export enum TicketType {
+  ADULT = "Adult",
+  STUDENT = "Student",
+  PENSIONER = "Pensioner",
```
Comment (Spuute) [inline shared/enums/ticket-type.enum.ts]: The same goes here, I think we should use SENIOR instead of PENSIONER

**Review by Spuute — CHANGES_REQUESTED** [review]
Overall I think you have done a great job! I have suggested some minor changes on some naming

File: api/src/controllers/booking.controller.ts (hunk 1 above)
Comment (adambrodin) [reply api/src/controllers/booking.controller.ts]: Changed name to reservation.

File: shared/enums/ticket-type.enum.ts (hunk 3 above)
Comment (adambrodin) [reply shared/enums/ticket-type.enum.ts]: Changed name to senior.

**Review by Spuute — APPROVED** [review]

**Pull request merged**